src/sagemaker/jumpstart/validators.py

```diff
@@ -0,0 +1,153 @@
+from __future__ import absolute_import
+from typing import Any, List, Optional
+from sagemaker.jumpstart.constants import JUMPSTART_DEFAULT_REGION_NAME
+
+from sagemaker.jumpstart.enums import HyperparameterValidationMode, VariableScope, VariableTypes
+from sagemaker.jumpstart import accessors as jumpstart_accessors
+from sagemaker.jumpstart.exceptions import JumpStartHyperparametersError
+from sagemaker.jumpstart.types import JumpStartHyperparameter
+
+
+def _validate_hyperparameter(
+    hyperparameter_name: str,
+    hyperparameter_value: Any,
+    hyperparameter_specs: List[JumpStartHyperparameter],
+):
+    """Perform low-level hyperparameter validation on single parameter.
+
+    Args:
+      hyperparameter_name (str): The name of the hyperparameter to validate.
+      hyperparameter_value (Any): The value of the hyperparemter to validate.
+      hyperparameter_specs (List[JumpStartHyperparameter]): List of ``JumpStartHyperparameter`` to
+        use when validating the hyperparameter.
+    """
+    hyperparameter_spec = [
+        spec for spec in hyperparameter_specs if spec.name == hyperparameter_name
+    ]
+    if len(hyperparameter_spec) == 0:
+        raise JumpStartHyperparametersError(
+            f"Unable to perform validation -- cannot find hyperparameter '{hyperparameter_name}' in model specs."
+        )
+    hyperparameter_spec = hyperparameter_spec[0]
+
+    if hyperparameter_spec.type == VariableTypes.TEXT.value:
+        if type(hyperparameter_value) != str:
+            raise JumpStartHyperparametersError(
+                f"Expecting text valued hyperparameter to have string type."
+            )
+
+        if getattr(hyperparameter_spec, "options", None):
+            if hyperparameter_value not in hyperparameter_spec.options:
+                raise JumpStartHyperparametersError(
+                    f"Hyperparameter '{hyperparameter_name}' must have one of the following values: "
+                    ", ".join(hyperparameter_spec.options)
+                )
+
+    # validate numeric types
+    if hyperparameter_spec.type in [VariableTypes.INT.value, VariableTypes.FLOAT.value]:
+        try:
+            numeric_hyperparam_value = float(hyperparameter_value)
+        except ValueError:
+            raise JumpStartHyperparametersError(
+                f"Hyperparameter '{hyperparameter_name}' must be numeric type ('{hyperparameter_value}')."
+            )
+
+        if hyperparameter_spec.type == VariableTypes.INT.value:
+            hyperparameter_value_str = str(hyperparameter_value)
+            start_index = 0
+            if hyperparameter_value_str[0] in ["+", "-"]:
+                start_index = 1
+            if not hyperparameter_value_str[start_index:].isdigit():
+                raise JumpStartHyperparametersError(
+                    f"Hyperparameter '{hyperparameter_name}' must be integer type ('{hyperparameter_value}')."
+                )
```

Comment:
would this be simpler:

```py
try:
    int_value = int(hyperparameter_value)
except ValueError:
   raise JumpStartHyperparametersError(f"...")
```

Reply:
That won't work, because floats can be serialized into ints.

Reply:
Wouldn't the training script arg [parser](https://github.com/aws/sagemaker-model-hub/blob/ffc4ec8750b48845a98e23c94d507fe61e30f4a6/src/sagemakermodelhub/scripts/pytorch/transfer_learning/eqa/transfer_learning.py#L35) take care of the conversion in such a case?